classes/Tools.php

```diff
@@ -809,7 +809,10 @@ protected static function getContextLocale(Context $context)
 
         /** @var LocaleRepository $localeRepository */
         $localeRepository = $container->get(self::SERVICE_LOCALE_REPOSITORY);
-        $locale = $localeRepository->getLocale((string) $context->language->locale);
+        $locale = !empty($context->language->locale) ?
+            $context->language->locale :
+            $context->language->language_code;
+        $locale = $localeRepository->getLocale((string) $locale);
```

Comment:
Can it be other thing than a string? why not casting in the second patch?

Reply:
That's a good question, I guess it was for handling a `null` value, but an empty string or `null` will both trigger an error.